integrationTests/multiShard/endOfEpoch/executingEpochChange_test.go

```diff
@@ -94,4 +98,36 @@ func TestEndOfEpochChangeWithoutTransactionInMultiShardedEnvironment(t *testing.
 	}
 
 	//////////------ verify last added shardheaders in meta are with new epoch
+	for _, node := range nodes {
+		if node.ShardCoordinator.SelfId() != sharding.MetachainShardId {
+			continue
+		}
+
+		currentMetaHdr, ok := node.BlockChain.GetCurrentBlockHeader().(*block.MetaBlock)
+		if !ok {
+			continue
+		}
+
+		shardHDrStorage := node.Storage.GetStorer(dataRetriever.BlockHeaderUnit)
```

Comment:
We should definitely build a service layer for these "getters" from blockchain/datapool/storage.